ipa-core/src/test_fixture/ipa.rs

```diff
@@ -253,7 +259,45 @@ pub async fn test_oprf_ipa<F>(
 
     //TODO(richaj): To be removed once the function supports non power of 2 breakdowns
     let _ = result.split_off(expected_results.len());
-    assert_eq!(result, expected_results);
+
+    match dp_params {
+        DpMechanism::NoDp => {
+            assert_eq!(result, expected_results);
+        }
+        DpMechanism::Binomial { epsilon } => {
+            let per_user_credit_cap_f64 = f64::from(config.per_user_credit_cap);
+            let noise_params = NoiseParams {
+                epsilon,
+                delta: 1e-6,
+                ell_1_sensitivity: per_user_credit_cap_f64,
+                ell_2_sensitivity: per_user_credit_cap_f64,
+                ell_infty_sensitivity: per_user_credit_cap_f64,
+                dimensions: 256.0, // matches hard coded dimension in oprf_ipa.rs/execute
+                ..Default::default()
+            };
+            let (mean, std) = crate::protocol::dp::noise_mean_std(&noise_params);
+
+            assert_eq!(result.len(), expected_results.len());
+            // would like to use the simpler version below but clippy and compiler disagree on
+            // importing zip and abs
```

Comment:
what's the error here? I am seeing `&sample` so that may not work, but `sample` will be a reference. If you want an owned value, you could call `into_iter` instead

Reply:
played with it a bit more and I think it is working now.